lua/lazyvim/plugins/extras/lang/python.lua

```diff
@@ -0,0 +1,65 @@
+return {
+  {
+    "nvim-treesitter/nvim-treesitter",
+    opts = function(_, opts)
+      if type(opts.ensure_installed) == "table" then
+        vim.list_extend(opts.ensure_installed, { "ninja", "python", "rst", "toml" })
+      end
+    end,
+  },
+  {
+    "neovim/nvim-lspconfig",
+    opts = {
+      servers = {
+        pyright = {},
+        ruff_lsp = {},
+      },
+    },
+    setup = {
+      ruff_lsp = function()
+        require("lazyvim.util").on_attach(function(client, _)
+          if client.name == "ruff_lsp" then
+            -- Disable hover in favor of Pyright
+            client.server_capabilities.hoverProvider = false
+          end
+        end)
+      end,
+    },
+  },
+  {
+    "nvim-neotest/neotest",
+    optional = true,
+    dependencies = {
+      "nvim-neotest/neotest-python",
+    },
+    opts = {
+      adapters = {
+        ["neotest-python"] = {
+          -- Here you can specify the settings for the adapter, i.e.
+          -- runner = "pytest",
+          -- python = ".venv/bin/python",
+        },
+      },
+    },
+  },
+  {
+    "mfussenegger/nvim-dap",
+    dependencies = {
+      "mfussenegger/nvim-dap-python",
+      -- stylua: ignore
+      keys = {
+        { "<leader>dPt", function() require('dap-python').test_method() end, desc = "Debug Method" },
+        { "<leader>dPc", function() require('dap-python').test_class() end, desc = "Debug Class" },
+      },
+      config = function()
+        local path = require("mason-registry").get_package("debugpy"):get_install_path()
+        require("dap-python").setup(path .. "/venv/bin/python")
```

Comment:
I've tried this setup too, but it doesn't work for me. All I get is that "Python exited with 1".
Does this work for you?

If you get this to work, you might want to have Mason install debugpy too as part of this config.

This kind of setup is a little devious, as you might have Python 3.11 install debugpy with Mason but you're working on a project with activated virtual environment in e.g. 3.7. Then this won't work either as the interpreter versions must match.

I've simply resorted to installing debugpy in each project of mine. Not great, but the only way I got this to work reliably.

Reply:
Hmm, okay so it might have something to do with that I've got a conflicting setup, [here](https://github.com/fredrikaverpil/dotfiles/blob/main/nvim-lazyvim/lua/plugins/debug.lua#L24-L48) as described [here](https://github.com/mfussenegger/nvim-dap/wiki/Debug-Adapter-installation#python).

Reply:
> Does this work for you?

Yes. The `debugpy` is automatically installed.

Reply:
I think you can do away with this completely, and instead make sure Mason installs debugpy. Then you'll get debugpy from mason automatically.